gateway/app/main/codes.py

```diff
@@ -1,18 +1,44 @@
-class GRID_MSG(object):
-    TYPE_FIELD = "type"
-    DATA_FIELD = "data"
+class MSG_FIELD:
+    TYPE = "type"
+    DATA = "data"
+    WORKER_ID = "worker_id"
+    MODEL = "model"
+    ALIVE = "alive"
+
+
+class CONTROL_EVENTS(object):
     SOCKET_PING = "socket-ping"
-    GET_PROTOCOL = "get-protocol"
+
+
+class WEBRTC_EVENTS(object):
     PEER_LEFT = "webrtc: peer-left"
     INTERNAL_MSG = "webrtc: internal-message"
     JOIN_ROOM = "webrtc: join-room"
 
 
+class FL_EVENTS(object):
+    REPORT = "federated/report"
+    AUTHENTICATE = "federated/authenticate"
+    CYCLE_REQUEST = "federated/cycle-request"
+
+
+class CYCLE(object):
+    STATUS = "status"
+    KEY = "request_key"
+    PING = "ping"
+    DOWNLOAD = "download"
+    UPLOAD = "upload"
+    VERSION = "version"
+    PLANS = "plans"
+    PROTOCOLS = "protocols"
+    CLIENT_CONFIG = "client_config"
+    SERVER_CONFIG = "server_config"
+    TIMEOUT = "timeout"
+    DIFF = "diff"
+    AVG_PLAN = "averaging_plan"
+    SECURE_AGG_PROTOCOL = "secure_agg_protocol"
```

Comment:
See my comment on "training plans" below.

It's worth noting that:
1. There may not be any protocol
2. There may be one protocol
3. There may be multiple protocols
4. Not all protocols will perform secure aggregation! It's entirely possible to have a protocol simply for validation.

DO NOT call it the "secure aggregation protocol" - this might be the most likely scenario, but it's not a rule. :)

Reply:
Done! :)